internal/storage/auth/sql/store.go

```diff
@@ -238,6 +238,41 @@ func (s *Store) ListAuthentications(ctx context.Context, req *storage.ListReques
 	return
 }
 
+func (s *Store) adaptError(fmtStr string, err *error) {
+	if *err != nil {
+		*err = fmt.Errorf(fmtStr, s.driver.AdaptError(*err))
+	}
+}
+
+// DeleteAuthentications attempts to delete one or more Authentication instances from the backing store.
+// Use auth.DeleteByID to construct a request to delete a single Authentication by ID string.
+// Use auth.DeleteByMethod to construct a request to delete 0 or more Authentications by Method and optional expired before constraint.
+func (s *Store) DeleteAuthentications(ctx context.Context, req *storageauth.DeleteAuthenticationsRequest) (err error) {
+	defer s.adaptError("deleting authentications: %w", &err)
+
+	if err := req.Valid(); err != nil {
+		return err
+	}
+
+	query := s.builder.
+		Delete("authentications")
+
+	if req.ID != nil {
+		query = query.Where(sq.Eq{"id": req.ID})
+	}
+
+	if req.Method != nil {
+		query = query.Where(sq.Eq{"method": req.Method.Method})
+		if req.Method.ExpiredBefore != nil {
```

Comment:
just to confirm, we're only allowing to delete by expiry if the method is provided? are you thinking this is a 'safety' check to prevent unintended deletes?

Reply:
That is a good question. What is this actually preventing?
I think I may be hyper focussed here on the two use-cases:
1. Delete an individual
2. Clean-up process, removing expired auths of a particular type.

But at the storage layer we could just keep this fairly generic. Three optional predicates:
1. Matches ID
2. Matches method
3. ExpiredAt is before supplied timestamp

I will break that out 👍